components/centraldashboard/app/profile_route_handlers.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ * This file contains the middleware used by the server to:
+ * 
+ * - Enrichen the request object with user / profile context
+ * - Redirect users to registration page if they have no profile
+ */
+import {Request, Response, NextFunction} from 'express';
+import {DefaultApi, Binding as WorkgroupBinding} from './clients/profile_controller';
+
+
+const {env} = process;
+const IAP_HEADER = env.IAP_HEADER || 'X-Goog-Authenticated-User-Email';
+const IAP_PREFIX = env.IAP_PREFIX || 'accounts.google.com:';
+
+// tslint:disable-next-line: no-any
+const G = global as any;
+G.USER_CACHE = {} || G.USER_CACHE;
+
+export interface AuthObject {
+    [iapHeader: string]: string;
+}
+
+export class UserData {
+    _req: Request;
```

Comment:
*nit*: Be consistent with property names. Since Typescript does enforce visibility, I don't believe the underscore prefix convention is followed. Also, for properties assigned in the constructor you can use the shorthand form to assign them.

https://kendaleiv.com/typescript-constructor-assignment-public-and-private-keywords/

Reply:
Done!